docs/operators.md

```diff
@@ -58,37 +63,38 @@ func main() {
 	kubeConfig := getKubeConfig()
 	
 	// Create a new operator
-	op := operator.New()
-	
-	// Create a controller to add to the operator
-	controller := operator.NewInformerController()
-	
-	// Create an informer to add to the controller
-	// MyTypeCustomResource is the generated CustomResource from schema_management.md
-	informer, err := operator.NewInformerFor(kubeConfig, MyTypeCustomResource, operator.NamespaceAll)
-	if err != nil {
-		// Do something with the error
-		panic(err)
-    }
-	
-	controller.AddInformer(informer)
-	
+	op, err := simple.NewOperator(simple.OperatorConfig{
+		Name:       "issue-operator",
+		KubeConfig: kubeConfig.RestConfig,
+		Metrics: simple.MetricsConfig{
+			Enabled: true,
+		},
+		Tracing: simple.TracingConfig{
+			Enabled: true,
+			OpenTelemetryConfig: simple.OpenTelemetryConfig{
+				Host:        cfg.OTelConfig.Host,
+				Port:        cfg.OTelConfig.Port,
+				ConnType:    simple.OTelConnType(cfg.OTelConfig.ConnType),
+				ServiceName: cfg.OTelConfig.ServiceName,
+			},
+		},
+		ErrorHandler: func(ctx context.Context, err error) {
+			logging.FromContext(ctx).Error(err.Error())
+		},
+	})
+
 	// Create a watcher which prints some lines when a resource is added
-	// We'll use an opinionated watcher, as that's best practice if you don't need to do anything fancy
-	watcher, err := operator.NewOpinionatedWatcher(kubeConfig, MyTypeCustomResource)
-	if err != nil {
-		// Do something with the error
-		panic(err)
-	}
-	watcher.AddFunc = func(ctx context.Context, object runtime.Object) error {
-		fmt.Println("Hey, a resource got added!")
-		return nil
+	watcher := simple.Watcher{
```

Comment:
@IfSentient should we recommend using Reconcilers instead of Watchers by default, or do you think it doesn't matter?

Reply:
I think at the moment it doesn't matter--kubernetes uses watchers for everything, so that might be a more familiar paradigm for some. Depending on the direction we take informers, reconcilers may be a better UX in the future (for example, if we have an informer that doesn't contain a cache, then you won't have a previous version of an object in a watcher's `Update` method), but at the moment both watchers and reconcilers are using the exact same [AddEventHandler](https://github.com/grafana/grafana-app-sdk/blob/main/operator/informer_kubernetes.go#L121) logic in the kubernetes SharedIndexInformer.

Reply:
For context, I was basing my assumption on the fact that the recommended way of writing controllers in "vanilla" k8s is https://github.com/kubernetes-sigs/controller-runtime, which only uses the reconciler pattern.

Reply:
Mmh it might make sense to use reconciler pattern here just to keep stuff as simple as possible? 
We can then add a link to the example folder, and might I suggest having three/four examples there:
- Watcher / Reconciler using `simple` package;
- Opinionated Watcher / Reconciler using `operator` package;

wdyt?

Reply:
SGTM 👍

Reply:
I'll use the reconciler pattern here then, and I'll work on the `examples` package in a followup 👍🏼

Reply:
Done!